features/admin.actions.v1/pages/action-configuration-page.tsx

```diff
@@ -0,0 +1,428 @@
+import { Checkbox, CheckboxProps, Grid } from "semantic-ui-react";
+import changeActionStatus from "../api/change-action-status";
+import deleteAction from "../api/delete-action";
+import useGetActionsByType from "../api/use-get-actions-by-type";
+import { ActionConfigForm } from "../components/action-config-form";
+import { ActionsConstants } from "../constants/actions-constants";
+import { ActionConfigFormPropertyInterface } from "../models/actions";
+import "./action-configuration-page.scss";
+
+/**
+ * Props for the Action Configuration page.
+ */
+type ActionConfigurationPageInterface = IdentifiableComponentInterface;
+
+export const ActionConfigurationPage: FunctionComponent<ActionConfigurationPageInterface> = ({
+    [ "data-componentid" ]: _componentId = "action-configuration-page"
+}: ActionConfigurationPageInterface): ReactElement => {
+
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features.actions);
+
+    const [ isOpenRevertConfigModal, setOpenRevertConfigModal ] = useState<boolean>(false);
+    const [ isActive, setIsActive ] = useState<boolean>(false);
+    const [ showCreateForm, setShowCreateForm ] = useState<boolean>(false);
+
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+    const { getLink } = useDocumentation();
+
+    const hasActionUpdatePermissions: boolean = useRequiredScopes(featureConfig?.actions?.scopes?.update);
+
+    const actionTypeApiPath: string = useMemo(() => {
+        const path: string[] = history.location.pathname.split("/");
+        const actionType: string = path[path.length - 1];
+
+        switch (actionType) {
+            case ActionsConstants.ACTION_TYPES.PRE_ISSUE_ACCESS_TOKEN.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_ISSUE_ACCESS_TOKEN.getApiPath();
+            case ActionsConstants.ACTION_TYPES.PRE_UPDATE_PASSWORD.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_UPDATE_PASSWORD.getApiPath();
+            case ActionsConstants.ACTION_TYPES.PRE_UPDATE_PROFILE.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_UPDATE_PROFILE.getApiPath();
+            case ActionsConstants.ACTION_TYPES.PRE_REGISTRATION.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_REGISTRATION.getApiPath();
+            default:
+                return null;
+        }
+    }, [ history.location.pathname ]);
+
+    const {
+        data: actions,
+        error: actionsFetchRequestError,
+        isLoading: isActionsLoading,
+        mutate: mutateActions
+    } = useGetActionsByType(actionTypeApiPath);
+
+    const isLoading: boolean = isActionsLoading || !actions || !Array.isArray(actions);
+
+    const actionInitialValues: ActionConfigFormPropertyInterface = useMemo(() => {
+        if (actions) {
+            return {
+                authenticationType: actions[0]?.endpoint?.authentication?.type.toString(),
```

Comment:
Shall we check the length of the actions array as well since this could break if we get an empty [] action array.

Reply:
It gives undefined for actions[0] if the actions array is empty. And it is expected

Reply:
The suggestion is to check the action array length in line 102 to prevent this.